panther_utils/include/panther_utils/ros_utils.hpp

```diff
@@ -0,0 +1,95 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PANTHER_UTILS__ROS_UTILS_HPP_
+#define PANTHER_UTILS__ROS_UTILS_HPP_
+
+#include <chrono>
+
+#include <std_msgs/msg/header.hpp>
+
+namespace panther_utils::ros
+{
+
+/**
+ * Verifies the timestamp gap between two headers and throws an exception if the gap exceeds the
+ * maximum allowed gap.
+ *
+ * @param header_1 The first header containing the timestamp.
+ * @param header_2 The second header containing the timestamp.
+ * @param max_timestamp_gap The maximum allowed timestamp gap in seconds.
+ * @throws `std::runtime_error` If either of the timestamps is not set or if the timestamp gap
+ * exceeds the maximum allowed gap.
+ */
+void VerifyTimestampGap(
+  const std_msgs::msg::Header & header_1, const std_msgs::msg::Header & header_2,
+  std::chrono::seconds max_timestamp_gap)
+{
+  if (header_1.stamp.sec == 0 || header_2.stamp.sec == 0) {
+    throw std::runtime_error("Timestamps are not set.");
+  }
+
+  const auto timestamp_1_ns = std::chrono::seconds(header_1.stamp.sec) +
+                              std::chrono::nanoseconds(header_1.stamp.nanosec);
+  const auto timestamp_2_ns = std::chrono::seconds(header_2.stamp.sec) +
+                              std::chrono::nanoseconds(header_2.stamp.nanosec);
```

Comment:
No nanosec checking.

```suggestion
  const auto timestamp_1_ns = std::chrono::seconds(header_1.stamp.sec) +
                              std::chrono::nanoseconds(header_1.stamp.nanosec);
  const auto timestamp_2_ns = std::chrono::seconds(header_2.stamp.sec) +
                              std::chrono::nanoseconds(header_2.stamp.nanosec);

  if (timestamp_1_ns.count() == 0 && timestamp_2_ns.count() == 0) {
    throw std::runtime_error("Timestamps are not set.");
  }
```

Reply:
That's because the `nanosec` part is not so crucial and may equal `0`. Anyway having calculated the "total" timestamp we can incorporate this guy to be verified 🆗